src/vivarium_public_health/metrics/utilities.py

```diff
@@ -442,6 +442,55 @@ def get_person_time_in_span(lived_in_span: pd.DataFrame, base_filter: QueryStrin
     return person_time
 
 
+def get_deaths(pop: pd.DataFrame, config: Dict[str, bool], sim_start: pd.Timestamp,
+               sim_end: pd.Timestamp, age_bins: pd.DataFrame) -> Dict[str, int]:
+    """Counts the number of deaths by cause.
+
+    The user is responsible for providing a default filter (e.g. only alive
+    people, or people susceptible to a particular disease).  Demographic
+    filters will be applied based on standardized configuration.
+
+    Parameters
+    ----------
+    pop
+        The population dataframe to be counted. It must contain sufficient
+        columns for any necessary filtering (e.g. the ``age`` column if
+        filtering by age).
+    config
+        A dict with ``by_age``, ``by_sex``, and ``by_year`` keys and
+        boolean values.
+    sim_start
+        The simulation start time.
+    sim_end
+        The simulation end time.
+    age_bins
+        A dataframe with ``age_group_start`` and ``age_group_end`` columns.
+
+    Returns
+    -------
+    deaths
+        A dictionary of output_key, death_count pairs where the output_key
+        represents a particular demographic subgroup.
+
+    """
+    base_filter = QueryString('alive == "dead" and cause_of_death == "{cause}"')
+    base_key = get_output_template(**config)
+    pop = clean_cause_of_death(pop)
+
+    causes = [c for c in pop.cause_of_death.unique()]
+    time_spans = get_time_iterable(config, sim_start, sim_end)
+
```

Comment:
these are so much cleaner and more readable with all the templating!

Reply:
Yeah.  There's a lot of duplicated logic though.  After I bring in all the functionality (and make a release) I'm going to spend a little time moving stuff around to try to get rid of code.